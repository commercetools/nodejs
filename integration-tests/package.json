{
  "private": true,
  "name": "@commercetools/integration-tests",
  "version": "0.1.22",
  "dependencies": {
    "@commercetools/api-request-builder": "^6.0.0",
    "@commercetools/category-exporter": "^4.0.0",
    "@commercetools/csv-parser-discount-code": "^3.0.0",
    "@commercetools/csv-parser-orders": "^3.0.0",
    "@commercetools/csv-parser-price": "^5.0.0",
    "@commercetools/csv-parser-state": "^4.0.0",
    "@commercetools/custom-objects-exporter": "^4.0.0",
    "@commercetools/custom-objects-importer": "^4.0.0",
    "@commercetools/customer-groups-exporter": "^4.0.0",
    "@commercetools/discount-code-exporter": "^5.0.0",
    "@commercetools/discount-code-generator": "^3.0.0",
    "@commercetools/discount-code-importer": "^4.0.0",
    "@commercetools/get-credentials": "^4.0.0",
    "@commercetools/http-user-agent": "3.0.0",
    "@commercetools/inventories-exporter": "^4.0.0",
    "@commercetools/personal-data-erasure": "^4.0.0",
    "@commercetools/platform-sdk": "^2.2.0",
    "@commercetools/price-exporter": "^4.0.0",
    "@commercetools/product-exporter": "^4.0.0",
    "@commercetools/product-json-to-csv": "^6.0.0",
    "@commercetools/product-json-to-xlsx": "^5.0.0",
    "@commercetools/resource-deleter": "^3.0.0",
    "@commercetools/sdk-auth": "^4.0.0",
    "@commercetools/sdk-client": "^3.0.0",
    "@commercetools/sdk-middleware-auth": "^7.0.0",
    "@commercetools/sdk-middleware-http": "^7.0.0",
    "@commercetools/sdk-middleware-logger": "3.0.0",
    "@commercetools/sdk-middleware-queue": "^3.0.0",
    "@commercetools/sdk-middleware-user-agent": "^3.0.0",
    "@commercetools/state-importer": "^4.0.0",
    "@commercetools/sync-actions": "^5.0.0",
    "bluebird": "^3.5.3",
    "common-tags": "^1.8.0",
    "csvtojson": "^2.0.8",
    "dotenv": "^8.0.0",
    "exceljs": "^3.0.0",
    "isuuid": "^0.1.0",
    "lodash.clonedeep": "^4.5.0",
    "lodash.omit": "^4.5.0",
    "lodash.zipobject": "^4.1.3",
    "mz": "^2.7.0",
    "node-fetch": "^2.6.7",
    "streamtest": "^2.0.0",
    "tmp": "^0.2.0",
    "unzipper": "^0.10.5"
  }
}
