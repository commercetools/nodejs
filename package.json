{
  "private": true,
  "license": "MIT",
  "engines": {
    "node": ">=6",
    "npm": ">=2"
  },
  "scripts": {
    "postinstall": "check-node-version --package --print && npm run bootstrap && cross-env NODE_ENV=development npm run build",
    "bootstrap": "lerna bootstrap",
    "check-updates": "./scripts/check-updates",
    "clean": "rm -rf coverage; lerna exec -- rimraf lib dist",
    "precommit": "lint-staged",
    "commitmsg": "commitlint -e $GIT_PARAMS",
    "commit": "git-cz",
    "build": "lerna run --parallel build",
    "lint": "jest --projects jest.eslint.config.js jest.flow.config.js",
    "lint:js": "jest --config jest.eslint.config.js",
    "test": "cross-env NODE_ENV=development jest --config jest.test.config.js",
    "test:package": "cross-env NODE_ENV=development ./scripts/test-package",
    "pretest:integration": "./scripts/install-npm-packages-for-integration-tests",
    "test:integration": "cross-env NODE_ENV=development jest --config integration-tests/jest.test.config.js",
    "format": "npm run format:js & npm run format:md",
    "format:js": "prettier --write '{packages,integration-tests,types}/**/*.js'",
    "format:md": "prettier --parser markdown --write '**/*.md'",
    "test:coverage": "npm run build && npm test -- --no-watchman --coverage",
    "test:ci": "npm run test -- --ci --no-watchman --maxWorkers=10",
    "test:coverage:ci": "npm run test:ci -- --coverage && codecov",
    "release": "cross-env NODE_ENV=cli npm run build && lerna publish --registry=https://registry.npmjs.org/ && npm run docs:publish",
    "docs:clean": "rimraf _book",
    "docs:prepare": "gitbook install",
    "docs:build": "npm run docs:prepare && gitbook build && cp logo/apple-touch-icon.png _book/gitbook/images/apple-touch-icon-precomposed-152.png && cp logo/favicon.ico _book/gitbook/images",
    "docs:watch": "npm run docs:prepare && gitbook serve",
    "docs:publish": "npm run docs:clean && npm run docs:build && ./scripts/publish-docs",
    "flow": "eslint packages",
    "package-readme-entry": "./scripts/package-readme-entry.sh"
  },
  "devDependencies": {
    "@commitlint/cli": "6.2.0",
    "@commitlint/config-conventional": "6.1.3",
    "babel-cli": "6.26.0",
    "babel-core": "6.26.3",
    "babel-eslint": "8.2.5",
    "babel-jest": "22.4.3",
    "babel-plugin-array-includes": "2.0.3",
    "babel-plugin-external-helpers": "6.22.0",
    "babel-plugin-transform-export-extensions": "6.22.0",
    "babel-plugin-transform-flow-strip-types": "6.22.0",
    "babel-plugin-transform-object-entries": "1.0.0",
    "babel-plugin-transform-object-rest-spread": "6.26.0",
    "babel-preset-env": "1.7.0",
    "babel-preset-stage-0": "6.24.1",
    "check-node-version": "3.2.0",
    "codecov": "3.0.4",
    "commitizen": "2.10.1",
    "common-tags": "1.8.0",
    "cross-env": "5.2.0",
    "cz-lerna-changelog": "1.2.1",
    "debug": "3.1.0",
    "eslint": "4.19.1",
    "eslint-config-airbnb": "16.1.0",
    "eslint-config-airbnb-base": "12.1.0",
    "eslint-config-prettier": "2.9.0",
    "eslint-formatter-pretty": "1.3.0",
    "eslint-plugin-flowtype": "2.49.3",
    "eslint-plugin-import": "2.13.0",
    "eslint-plugin-jest": "21.17.0",
    "eslint-plugin-jsx-a11y": "6.1.0",
    "eslint-plugin-prettier": "2.6.2",
    "eslint-plugin-react": "7.10.0",
    "flow-bin": "0.74.0",
    "gitbook-cli": "2.3.2",
    "gitbook-plugin-anchorjs": "1.1.1",
    "gitbook-plugin-edit-link": "2.0.2",
    "gitbook-plugin-ga": "1.0.1",
    "gitbook-plugin-github": "2.0.0",
    "gitbook-plugin-prism": "2.3.0",
    "husky": "0.14.3",
    "jest": "22.4.3",
    "jest-runner-eslint": "emmenko/jest-runner-eslint#nm-fix-rules-cli-option-built",
    "jest-runner-flowtype": "0.0.5",
    "jest-silent-reporter": "0.0.4",
    "lerna": "2.11.0",
    "lint-staged": "7.0.4",
    "npm-check-updates": "2.14.1",
    "prettier": "1.11.1",
    "resolve": "1.7.1",
    "rimraf": "2.6.2",
    "rollup": "0.57.1",
    "rollup-plugin-babel": "3.0.3",
    "rollup-plugin-commonjs": "8.4.1",
    "rollup-plugin-filesize": "1.5.0",
    "rollup-plugin-flow": "1.1.1",
    "rollup-plugin-includepaths": "0.2.2",
    "rollup-plugin-json": "2.3.0",
    "rollup-plugin-node-resolve": "3.3.0",
    "rollup-plugin-replace": "2.0.0",
    "rollup-plugin-uglify": "2.0.1",
    "rollup-watch": "4.3.1",
    "streamtest": "1.2.3"
  },
  "config": {
    "commitizen": {
      "path": "./node_modules/cz-lerna-changelog"
    }
  }
}
