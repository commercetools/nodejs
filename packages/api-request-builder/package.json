{
  "private": false,
  "publishConfig": {
    "access": "public"
  },
  "name": "@commercetools/api-request-builder",
  "version": "3.3.1",
  "description": "Helper functions to construct API requests URI for the commercetools platform in a declarative way.",
  "keywords": [
    "commercetools",
    "api",
    "uri",
    "builder",
    "request"
  ],
  "homepage": "https://commercetools.github.io/nodejs/",
  "bugs": "https://github.com/commercetools/nodejs/issues",
  "license": "MIT",
  "author": "Nicola Molinari <nicola.molinari@commercetools.com> (https://github.com/emmenko)",
  "main": "dist/commercetools-api-request-builder.cjs.js",
  "module": "dist/commercetools-api-request-builder.es.js",
  "browser": "dist/commercetools-api-request-builder.umd.js",
  "repository": {
    "type": "git",
    "url": "https://github.com/commercetools/nodejs.git"
  },
  "files": [
    "dist",
    "lib"
  ],
  "scripts": {
    "build": "cross-env npm run build:umd && npm run build:umd:min && npm run build:es && npm run build:cjs",
    "build:umd": "cross-env NODE_ENV=development rollup -c ../../rollup.config.js -f umd -n CommercetoolsApiRequestBuilder -i src/index.js -o dist/commercetools-api-request-builder.umd.js",
    "build:umd:min": "cross-env NODE_ENV=production rollup -c ../../rollup.config.js -f umd -n CommercetoolsApiRequestBuilder -i src/index.js -o dist/commercetools-api-request-builder.umd.min.js",
    "build:es": "cross-env NODE_ENV=development rollup -c ../../rollup.config.js -f es -n CommercetoolsApiRequestBuilder -i src/index.js -o dist/commercetools-api-request-builder.es.js",
    "build:cjs": "cross-env NODE_ENV=development rollup -c ../../rollup.config.js -f cjs -n CommercetoolsApiRequestBuilder -i src/index.js -o dist/commercetools-api-request-builder.cjs.js"
  }
}
