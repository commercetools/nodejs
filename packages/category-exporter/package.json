{
  "private": false,
  "publishConfig": {
    "access": "public"
  },
  "name": "@commercetools/category-exporter",
  "version": "3.0.22",
  "description": "Exports categories from the commercetools platform.",
  "main": "lib/main.js",
  "bin": {
    "category-exporter": "bin/category-exporter.js"
  },
  "scripts": {
    "prebuild": "rimraf lib/**",
    "build": "babel src --out-dir lib --config-file '../../babel.config.js'"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/commercetools/nodejs/tree/master/packages/category-exporter"
  },
  "homepage": "https://commercetools.github.io/nodejs/cli/category-exporter.html",
  "bugs": "https://github.com/commercetools/nodejs/issues",
  "keywords": [
    "commercetools",
    "category",
    "categories",
    "json",
    "export",
    "cli"
  ],
  "author": {
    "name": "Babajide Williams",
    "email": "babajide.williams@commercetools.com"
  },
  "license": "MIT",
  "dependencies": {
    "@commercetools/api-request-builder": "^5.6.3",
    "@commercetools/get-credentials": "^3.0.17",
    "@commercetools/sdk-client": "^2.1.2",
    "@commercetools/sdk-middleware-auth": "^5.1.6",
    "@commercetools/sdk-middleware-http": "^6.0.11",
    "@commercetools/sdk-middleware-user-agent": "^2.1.5",
    "node-fetch": "^2.3.0",
    "pino": "^6.0.0",
    "pino-pretty": "^4.0.0",
    "pretty-error": "^3.0.0",
    "yargs": "^15.0.0"
  },
  "devDependencies": {
    "streamtest": "2.0.0"
  }
}
