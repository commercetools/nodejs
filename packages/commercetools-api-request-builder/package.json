{
  "private": true,
  "name": "@commercetools/api-request-builder",
  "version": "1.0.0",
  "description": "Helper functions to construct API requests URI for the commercetools platform in a declarative way.",
  "keywords": [
    "commercetools",
    "api",
    "uri",
    "builder",
    "request"
  ],
  "homepage": "https://github.com/commercetools/nodejs",
  "bugs": "https://github.com/commercetools/nodejs/issues",
  "license": "MIT",
  "author": "",
  "main": "./lib/index.js",
  "repository": {
    "type": "git",
    "url": "https://github.com/commercetools/nodejs.git"
  },
  "files": [
    "lib"
  ]
}
