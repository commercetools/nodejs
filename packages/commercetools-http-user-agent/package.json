{
  "private": false,
  "publishConfig": {
    "access": "public"
  },
  "name": "@commercetools/http-user-agent",
  "version": "0.1.0",
  "description": "Creates a proper HTTP User-Agent",
  "keywords": [
    "commercetools",
    "http",
    "user-agent"
  ],
  "homepage": "https://commercetools.github.io/nodejs/",
  "bugs": "https://github.com/commercetools/nodejs/issues",
  "license": "MIT",
  "author": "Nicola Molinari <nicola.molinari@commercetools.com> (https://github.com/emmenko)",
  "main": "lib/index.js",
  "repository": {
    "type": "git",
    "url": "https://github.com/commercetools/nodejs.git"
  },
  "files": [
    "dist",
    "lib"
  ],
  "scripts": {
    "dist": "npm run build:umd && npm run build:umd:min",
    "build:umd": "cross-env cross-env NODE_ENV=development webpack --config ../../webpack.config.js --output-library CommercetoolsHttpUserAgent --output-library-target umd src/index.js dist/commercetools-http-user-agent.js",
    "build:umd:min": "cross-env cross-env NODE_ENV=production webpack --config ../../webpack.config.js --output-library CommercetoolsHttpUserAgent --output-library-target umd src/index.js dist/commercetools-http-user-agent.min.js"
  }
}
