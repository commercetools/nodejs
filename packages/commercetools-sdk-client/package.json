{
  "private": true,
  "name": "@commercetools/sdk-client",
  "version": "1.0.0",
  "description": "Node.js SDK Client for usage of commercetools platform API",
  "keywords": [
    "commercetools",
    "sdk",
    "api",
    "client"
  ],
  "homepage": "https://github.com/commercetools/nodejs",
  "bugs": "https://github.com/commercetools/nodejs/issues",
  "license": "MIT",
  "author": "",
  "main": "./lib/index.js",
  "repository": {
    "type": "git",
    "url": "https://github.com/commercetools/nodejs.git"
  },
  "devDependencies": {
    "@commercetools/sdk-middleware-auth": "*",
    "@commercetools/sdk-middleware-http": "*",
    "nock": "~9.0.2"
  }
}
