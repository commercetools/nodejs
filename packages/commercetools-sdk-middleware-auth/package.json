{
  "private": true,
  "name": "@commercetools/sdk-middleware-auth",
  "version": "1.0.0",
  "description": "",
  "keywords": [
    "commercetools",
    "nodejs",
    "sdk",
    "middleware",
    "auth",
    "oauth"
  ],
  "homepage": "https://github.com/commercetools/nodejs",
  "bugs": "https://github.com/commercetools/nodejs/issues",
  "license": "MIT",
  "author": "",
  "main": "",
  "repository": {
    "type": "git",
    "url": "https://github.com/commercetools/nodejs.git"
  },
  "dependencies": {
    "isomorphic-fetch": "2.2.1"
  },
  "devDependencies": {
    "nock": "~9.0.2"
  }
}
