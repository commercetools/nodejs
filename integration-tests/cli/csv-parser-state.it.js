import path from 'path'
import isuuid from 'isuuid'
import { exec } from 'mz/child_process'
import { version } from '@commercetools/csv-parser-state/package.json'

let projectKey
if (process.env.CI === 'true') projectKey = 'state-import-integration-test'
else projectKey = process.env.npm_config_projectkey

describe('CSV and CLI Tests', () => {
  const binPath = './integration-tests/node_modules/.bin/csv-parser-state'

  describe('CLI basic functionality', () => {
    it('should print usage information given the help flag', async () => {
      const [stdout, stderr] = await exec(`${binPath} --help`)
      expect(stderr).toBeFalsy()
      expect(stdout).toMatchSnapshot()
    })

    it('should print the module version given the version flag', async () => {
      const [stdout, stderr] = await exec(`${binPath} --version`)
      expect(stderr).toBeFalsy()
      expect(stdout).toBe(`${version}\n`)
    })
  })

  describe('Parse', () => {
    describe('without transitions', () => {
      const statesCsv = path.join(
        __dirname,
        './helpers/statesWithoutTransitions.csv'
      )
      // No projectkey ensures the API is not called
      it('should parse states without projectKey', async () => {
        const [stdout, stderr] = await exec(`${binPath} --input ${statesCsv}`)
        expect(stderr).toBeFalsy()
        expect(JSON.parse(stdout)).toHaveLength(2)
        expect(stdout).toMatchSnapshot()
      })
    })

    describe('with transitions', () => {
      const statesCsv = path.join(
        __dirname,
        '../../packages/csv-parser-state/test/helpers/sampleStates.csv'
      )
      describe('without `projectkey`', () => {
        describe('without `continueOnProblems`', () => {
          it('should stop parsing and return error', async () => {
            await expect(
              exec(`${binPath} --input ${statesCsv}`)
            ).rejects.toThrow(/No project defined/)
          })
        })

        describe('with `continueOnProblems`', () => {
          it('should skip rows having transition', async () => {
            const [stdout, stderr] = await exec(
              `${binPath} --input ${statesCsv} --continueOnProblems`
            )
            expect(stderr).toBeFalsy()
            expect(JSON.parse(stdout)).toHaveLength(3)
            expect(stdout).toMatchSnapshot()
          })
        })
      })

      describe('with `projectKey`', () => {
        it('should resolve transition and parse states', async () => {
          const [stdout, stderr] = await exec(
            `${binPath} -p ${projectKey} --input ${statesCsv}`
          )
          expect(stderr).toBeFalsy()
          const parsedStates = JSON.parse(stdout)
          expect(parsedStates).toHaveLength(4)
          // ids cannot be verified with snapshots as they are dynamic,
          // so verify separately and remove
          expect(isuuid(parsedStates[1].transitions[0].id)).toBe(true)
          delete parsedStates[1].transitions[0].id
          expect(parsedStates).toMatchSnapshot()
        })
      })
    })
  })
})
