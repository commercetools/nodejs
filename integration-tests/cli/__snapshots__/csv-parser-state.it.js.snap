// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSV and CLI Tests CLI basic functionality should print usage information given the help flag 1`] = `
"
Usage: csv-parser-state [options]
Package to parse states from CSV to JSON

Options:
  --help                     Show help                                 [boolean]
  --version                  Show version number                       [boolean]
  --projectKey, -p           API project key                            [string]
  --apiUrl                   The host URL of the HTTP API service
                             [string] [default: \\"https://api.commercetools.com\\"]
  --authUrl                  The host URL of the OAuth API service
                            [string] [default: \\"https://auth.commercetools.com\\"]
  --accessToken              CTP client access token
                             Required scopes: ['view_orders']           [string]
  --input, -i                Path to CSV file.                [default: \\"stdin\\"]
  --output, -o               Path to output JSON file
                                                    [string] [default: \\"stdout\\"]
  --delimiter, -d            Used CSV delimiter.                  [default: \\",\\"]
  --multiValueDelimiter, -m  Used CSV delimiter in multiValue fields.
                                                                  [default: \\";\\"]
  --continueOnProblems, -c   Flag if parsing should continue if module
                             encounters an error.     [boolean] [default: false]
  --logLevel                 Logging level: error, warn, info or debug
                                                      [string] [default: \\"info\\"]
  --prettyLogs               Pretty print logs to the terminal         [boolean]
  --logFile                  Path to file where logs should be saved
                                      [string] [default: \\"csv-parser-state.log\\"]
"
`;

exports[`CSV and CLI Tests Parse with transitions with \`projectKey\` should resolve transition and parse states 1`] = `
Array [
  Object {
    "description": Object {
      "en": "beschreibung",
    },
    "initial": true,
    "key": "state-key-1",
    "name": Object {
      "en": "Will's state",
    },
    "type": "OrderState",
  },
  Object {
    "description": Object {
      "en": "uniqueness of states",
    },
    "initial": false,
    "key": "state-key-2",
    "name": Object {
      "en": "tester",
    },
    "transitions": Array [
      Object {
        "typeId": "state",
      },
    ],
    "type": "ProductState",
  },
  Object {
    "description": Object {
      "en": "only used when some state is complete",
    },
    "initial": false,
    "key": "state-key-3",
    "name": Object {
      "en": "completion",
    },
    "type": "LineItemState",
  },
  Object {
    "description": Object {
      "en": "this state should be auto generated",
    },
    "initial": true,
    "key": "state-key-4",
    "name": Object {
      "en": "auto gen",
    },
    "type": "PaymentState",
  },
]
`;

exports[`CSV and CLI Tests Parse with transitions without \`projectkey\` with \`continueOnProblems\` should skip rows having transition 1`] = `
"[
{\\"key\\":\\"state-key-1\\",\\"type\\":\\"OrderState\\",\\"name\\":{\\"en\\":\\"Will's state\\"},\\"initial\\":true,\\"description\\":{\\"en\\":\\"beschreibung\\"}}
,
{\\"key\\":\\"state-key-3\\",\\"type\\":\\"LineItemState\\",\\"name\\":{\\"en\\":\\"completion\\"},\\"initial\\":false,\\"description\\":{\\"en\\":\\"only used when some state is complete\\"}}
,
{\\"key\\":\\"state-key-4\\",\\"type\\":\\"PaymentState\\",\\"name\\":{\\"en\\":\\"auto gen\\"},\\"initial\\":true,\\"description\\":{\\"en\\":\\"this state should be auto generated\\"}}
]
"
`;

exports[`CSV and CLI Tests Parse without transitions should parse states without projectKey 1`] = `
"[
{\\"key\\":\\"state-key-1\\",\\"type\\":\\"OrderState\\",\\"name\\":{\\"en\\":\\"Will's state\\"},\\"initial\\":true,\\"description\\":{\\"en\\":\\"beschreibung\\"}}
,
{\\"key\\":\\"state-key-2\\",\\"type\\":\\"ProductState\\",\\"name\\":{\\"en\\":\\"tester\\"},\\"initial\\":false,\\"description\\":{\\"en\\":\\"uniqueness of states\\"}}
]
"
`;
